public/types.ts

```diff
@@ -17,4 +19,18 @@ export interface QueryEnhancementsPluginSetupDependencies {
 
 export interface QueryEnhancementsPluginStartDependencies {
   data: DataPublicPluginStart;
+  dataSource?: DataSourcePluginStart;
+}
+
+export interface Connection {
+  id: string;
+  title: string;
+  endpoint?: string;
+  installedPlugins?: string[];
+  auth?: any;
```

Comment:
should this connection type be picked from https://github.com/opensearch-project/OpenSearch-Dashboards/blob/e751ed6526428f41763600903b897cc00e55cba6/src/plugins/data_source/common/data_sources/types.ts#L15-L18

Reply:
yeah it should. the original approach i wanted to go with is not requiring any imports from the data source plugin in case it didn't get loaded in browser (data_source.enabled: false) because i dont think they have any lazy loading over there.

i think my goal here is even if we want to take the work for the flint datasources stuff can live in this repo but extend what they have in data sources.

where we can properly lazy loaded and make it a lot easier to use and then eventually unify it with the ds mgmt plugin

Reply:
i see. i thought it wouldn't matter for types and `import`/`import type {xxx` would be removed when transpiled to js